flink-table/flink-table-common/src/main/java/org/apache/flink/table/module/Module.java

```diff
@@ -36,17 +36,28 @@
 public interface Module {
 
     /**
-     * List names of all functions in this module.
+     * List names of all functions in this module. It excludes internal functions.
      *
      * @return a set of function names
      */
     default Set<String> listFunctions() {
         return Collections.emptySet();
     }
 
+    /**
+     * List names of all functions in this module. It can include internal functions.
+     *
+     * @return a set of function names
+     */
+    default Set<String> listFunctions(boolean includeInternal) {
```

Comment:
I'm not a fan of using a boolean to filter something in a `PublicEvolving` interface. Use an enum instead?

Reply:
In general, I don't understand why you want to expose the internal functions from the `Module`. Module is a public interface, and the change to the interface to list internal functions sounds like it's leaking some internal detail of how our operators behave. Have you considered implementing the resolution of internal functions at another level, e.g. within `FunctionCatalog`?

Reply:
The module system does not the concept of an internal function. Instead it is only able to hide certain functions during listing. I rephrased this part and split it into a separate commit.